meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start/compose-apps-register-images.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Register images of Apps at a docker store repository
+Before=docker.service
+Before=lmp-device-auto-register.service
+ConditionPathExists=/var/sota/reset-apps
+ConditionPathExists=!/var/sota/sql.db
+
+[Service]
+Type=oneshot
+RemainAfterExit=true
+ExecStart=/usr/bin/aklite-apps hack
```

Comment:
Hack?

Reply:
@mike-sul is hack the right argument here? If so, why is it called hack?

Reply:
Yes, the `hack` is a correct argument here. It's a nice name for a command isn't it? :).

For some reason, neither the TAR image format, that is used for loading of images into a dockerd's store, nor dockerd does NOT allow to "pin"/map a loaded image to a digested image reference (image URL with hash). Our solution (compose app, aklite) refers to images by digested reference. So, even though all required images are loaded into a dockerd's store, it (dockerd) communicates to Registry to fetch image manifest because it does not known whether a given image referenced  as e.g. hub.f.io/factory/foo/app@sha256:<hash> really exists in its store or not. Once it downloads an image manifest it gets an image config hash and checks whether it's already in the store, if so, only then it knows that the given image already loaded.
Obviously, such behavior is not appropriate for our "early app startup in offline mode" case. So, this command `hack` "tells" `dockerd` that prelaoded images referenced by digested reference already present in its store what makes `dockerd` to not do any connections to Registry. Specifically, the command adds required mapping (image uri -> image config hash) to  `/var/lib/docker/image/overlay2/repositories.json`.

It's kind of dockerd hacking hence the command is called `hack`. Having said that, I don't have a strong opinion about it and open to any suggestions on the command naming.

^^ @doanac FYI.

Reply:
looking at it now that a week has passed - i don't clearly remember what "hack" even does - so it's a bad name. I think we should name it better to help the next person that goes to debug this

Reply:
renamed the cmd to `register`.